File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    
    queue<string> c1;
    for(const string& s: cards1) c1.push(s);
    queue<string> c2;
    for(const string& s: cards2) c2.push(s);
    
    string answer = "Yes";
    
    for(const string& word: goal) {
        if (c1.front() == word) c1.pop();
        else if (c2.front() == word) c2.pop();
        else {
            answer = "No";
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        Queue<String> c1 = new LinkedList<>();
        for(String s: cards1) c1.add(s);
        Queue<String> c2 = new LinkedList<>();
        for(String s: cards2) c2.add(s);
        
        String answer = "Yes";
        for(String s: goal) {
            if (!c1.isEmpty() && c1.peek().equals(s)) c1.poll();
            else if (!c2.isEmpty() && c2.peek().equals(s)) c2.poll();
            else {
                answer = "No";
                break;
            }
        }
        
        return answer;
    }
}